docker/bin/diffdb

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+# compares the vanilla db schema of this docker image with the one of the
+# configured database server
+set -e
+source /usr/local/lib/utils.sh
+
+: "${DIFFDB_WORKDIR:=/tmp}"
+
+if [ ! -d "$DIFFDB_WORKDIR" ]
+then
+  log "working directory $DIFFDB_WORKDIR does not exist, exiting"
+  exit 1
+fi
+
+log "using $DIFFDB_WORKDIR as working directory"
+
+log dumping database schema from target server to $DIFFDB_WORKDIR/old_dump.sql ...
+pg_dump $DIFFDB_DUMP_OPTIONS -f $DIFFDB_WORKDIR/old_dump.sql
+pg_dump $DIFFDB_DUMP_OPTIONS -f $DIFFDB_WORKDIR/old_dump.sql
+log  dumping globals from target server to $DIFFDB_WORKDIR/old_globals.sql
+pg_dumpall -s --no-role-passwords --roles-only --clean --if-exists --file="$DIFFDB_WORKDIR/old_globals.sql"
+
+log "creating vanilla dump"
+create_vanilla_dump $DIFFDB_WORKDIR/vanilla_dump1.sql > $DIFFDB_WORKDIR/vanilla_dump1.log
+
+log "running first diff"
+apgdiff --drop-cascade --out-file $DIFFDB_WORKDIR/diff1.sql $DIFFDB_WORKDIR/old_dump.sql $DIFFDB_WORKDIR/vanilla_dump1.sql
+
+log "concatenating diffs"
+echo "-- GLOBALS $DIFFDB_WORKDIR/old_globals.sql -----------------" > $DIFFDB_WORKDIR/init2.sql
+egrep -v '^(CREATE|DROP) ROLE.* postgres;' $DIFFDB_WORKDIR/old_globals.sql >> $DIFFDB_WORKDIR/init2.sql
```

Comment:
a comment why this is needed would be helpful to me

Reply:
we do not want to drop / create the postgres user, since this would require a re-connect